csharp/src/Apache.Arrow.Flight.Sql/Client/FlightSqlClient.cs

```diff
@@ -0,0 +1,1191 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Apache.Arrow.Flight.Client;
+using Apache.Arrow.Types;
+using Arrow.Flight.Protocol.Sql;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+
+namespace Apache.Arrow.Flight.Sql.Client;
+
+public class FlightSqlClient
+{
+    private readonly FlightClient _client;
+
+    public FlightSqlClient(FlightClient client)
+    {
+        _client = client;
+    }
+
+    /// <summary>
+    /// Execute a SQL query on the server.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <param name="query">The UTF8-encoded SQL query to be executed.</param>
+    /// <param name="transaction">A transaction to associate this query with.</param>
+    /// <returns>The FlightInfo describing where to access the dataset.</returns>
+    public async Task<FlightInfo> ExecuteAsync(FlightCallOptions options, string query, Transaction? transaction = null)
+    {
+        transaction ??= Transaction.NoTransaction;
+
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (string.IsNullOrEmpty(query))
+        {
+            throw new ArgumentException($"Query cannot be null or empty: {nameof(query)}");
+        }
+
+        try
+        {
+            var prepareStatementRequest =
```

Comment:
I don't believe this is the proper way to implement execute.  There shouldn't be a need to create and mange prepare statements (for which this implementation leaks by not closing the statement).  This should use the `CommandStatementQuery` command to execute the query.